### ISSUE TITLE
iterating benedict_dictionary.items(), value is not benedict_type 

### ISSUE THREAD

**Issue by Loreton** [fabiocaccamo/python-benedict#333]
**Python version**
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux

**Package version**
python-benedict        0.32.0

**Current behavior (bug description)**
```
from benedict import benedict

likes = {
        "fruit": {
            "apple": "green", 
            "banana": "yellow",
            "raspberry": "red",
            }
        }


likes=benedict(likes, keyattr_enabled=True, keyattr_dynamic=False)

for k, v in likes.items():  
     print(k, type(v))
     print(k, type(likes[k]))
```
**Result**
```
fruit <class 'dict'>
fruit <class 'benedict.dicts.benedict'>
```
**Expected behavior**

The item value returned by the for loop is a **dict** and not a **benedict.dict** as it should be. This behavior is intentional or is it a bug?

```
fruit <class 'benedict.dicts.benedict'>
fruit <class 'benedict.dicts.benedict'>
```
Regards

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/fabiocaccamo) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/fabiocaccamo/python-benedict/issues/333">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/fabiocaccamo/python-benedict/issues/333/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/fabiocaccamo/python-benedict/issues/333/pledge.svg">
</picture>
</a>

**Comment by fabiocaccamo** [comment]
@Loreton the behaviour is unintentional, thank you for reporting it.

**Comment by fabiocaccamo** [comment]
@Loreton fixed in `0.32.1` version.
